### PR TITLE
Remove post delayed callbacks when view is detached from window

### PR DESCRIPTION
**Pull request by msasikanth**

### CODE REVIEW

**Comment by msasikanth** [comment]
Showing keyboard in `doOnAttach` action is not working perfectly in pin entry card view. Switching to the new `WindowInsetsControllerCompat` to show/hide the keyboard fixes this, but breaks showing keyboard in the instant search screen. So, for now, I just stuck with an approach that works and just removed the callback when the view is detached.

File: app/src/main/java/org/simple/clinic/widgets/Views.kt [hunk 1]
```diff
@@ -31,11 +31,13 @@ import timber.log.Timber
 import java.time.Duration
 
 fun EditText.showKeyboard() {
-  postDelayed({
-    this.requestFocus()
+  fun showKeyboard() {
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/widgets/Views.kt:35]: Can we rename this to something else? Seems weird to have a local function named the same as the enclosing function.

**Review by vinaysshenoy — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/simple/clinic/widgets/Views.kt (hunk 1 above)
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/widgets/Views.kt:35]: Yep, renaming it to `openKeyboard`

**Review by vinaysshenoy — APPROVED** [review]

**Pull request merged**